azurelinuxagent/ga/exthandlers.py

```diff
@@ -253,15 +258,22 @@ def run(self):
                       message=detailed_msg)
             return
 
+        # For extensions FastTrack, we can receive goal state through the VMArtifactsProfile blob
+        # For now, failing to read it isn't a critical failure. Even if storage is completely down
+        # we'll eventually receive it through Fabric goal state
+        self.get_artifacts_profile()
+
         try:
-            msg = u"Handle extensions updates for incarnation {0}".format(etag)
+            msg = u"Handle extensions updates for incarnation {0}".format(instantiation)
             logger.verbose(msg)
             # Log status report success on new config
             self.log_report = True
 
             if self.extension_processing_allowed():
-                self.handle_ext_handlers(etag)
-                self.last_etag = etag
+                sequence_number = None
+                if self.artifacts_profile is not None and conf.get_extensions_fast_track_enabled():
+                    sequence_number = self.artifacts_profile.get_sequence_number()
```

Comment:
```
sequence_number = None
if self.artifacts_profile is not None and conf.get_extensions_fast_track_enabled():
    sequence_number = self.artifacts_profile.get_sequence_number()
```
Does computing `sequence_number` help and used? Else push it into the `process_goal_state` method. Keeps it cleaner.

Reply:
Yes. It's used in the next line when processing the goal state.